x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceKey.java

```diff
@@ -63,4 +63,18 @@ public boolean equals(Object obj) {
     public String toString() {
         return CollectionUtils.isEmpty(keys) ? "NONE" : Arrays.toString(keys);
     }
+
+    @Override
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public int compareTo(SequenceKey o) {
+        for (int i = 0; i < this.keys.length && i < o.keys.length; i++) {
+            if (keys[i] instanceof Comparable) {
+                int result = ((Comparable) keys[i]).compareTo(o.keys[i]);
+                if (result != 0) {
+                    return result;
+                }
+            }
+        }
+        return this.keys.length - o.keys.length;
```

Comment:
Either use `this.keys` through out the method or just `keys; using both styles makes it hard to follow.